datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/prim_op.rs

```diff
@@ -53,10 +56,20 @@ where
     starting_value: T::Native,
 
     /// Track nulls in the input / filters
-    null_state: NullState,
+    null_state: NullStateAdapter,
 
     /// Function that computes the primitive result
     prim_fn: F,
+
+    /// Block size of current `GroupAccumulator` if exist:
+    ///   - If `None`, it means block optimization is disabled,
+    ///     all `group values`` will be stored in a single `Vec`
+    ///
+    ///   - If `Some(blk_size)`, it means block optimization is enabled,
```

Comment:
I wonder what you think about not keeping the code around to support both paths, but instead always used the blocked approach? It seems to me like in the long term if we try to support both paths it will be hard to ensure they are all tested and thus will result in subtle bugs. Keeping as few combinations as possible seems more maintainable to me